### PR TITLE
Add Alex's protect implementation

### PR DESCRIPTION
**Pull request by natanlao**
@cket 
jenkins, skip this please

### CODE REVIEW

File: protect/create.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    --sa $(pwd)/sa \
+    --output-dir $(pwd)/ \
+    --sort \
+    &>final.txt & cat final.txt
+
+exit 
+
+patients:
+    TEST:
+        tumor_dna_fastq_1 : S3://cgl-protect-data/ci_references/Tum_1.fq.gz
```
Comment (cket) [inline protect/create.sh]: @natanlao looks like there is test data here, when you get a chance can you try running it on a c3.8xlarge EC2 instance?

File: protect/Makefile [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Definitions
+docker_versions = 1.7.0 1.7.1 1.8.0 1.8.1 1.9.0 1.9.1 1.10.0 1.10.1 1.10.3
```
Comment (cket) [inline protect/Makefile]: Check out the Docker version range in https://github.com/BD2KGenomics/cgl-docker-lib/pull/214 and use those. Those are the docker versions that actually change the docker protocol version, which is what we care about.

File: protect/Dockerfile.template [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    git \
+    python-dev \
+    python-pip \
+    wget \
+    curl \
+    apt-transport-https \
+    ca-certificates
+
+# Get the Docker binary
+RUN wget https://get.docker.com/builds/Linux/x86_64/docker-DOCKERVER -O /usr/local/bin/docker
```
Comment (cket) [inline protect/Dockerfile.template]: Can you use the Docker download command from #214 and use it here? The newer version of docker changed the way they distribute binaries & this wont work for those new versions

**Comment by jvivian** [comment]
+1 for commit comments

**Comment by cket** [comment]
@natanlao I disabled jenkins testing of this PR for now since there aren't any tests for Protect yet - you can reenable it by editing the PR's description & removing the jenkins directive there.

**Comment by natanlao** [comment]
Stale.

**Pull request closed without merging**